src/data/value/mod.rs

```diff
@@ -36,14 +38,30 @@ pub enum Value {
     Null,
 }
 
+trait TryBinaryOperator {
+    type Rhs;
+
+    fn try_add(&self, rhs: &Self::Rhs) -> Result<Value>;
+    fn try_subtract(&self, rhs: &Self::Rhs) -> Result<Value>;
+    fn try_multiply(&self, rhs: &Self::Rhs) -> Result<Value>;
+    fn try_divide(&self, rhs: &Self::Rhs) -> Result<Value>;
+    fn try_modulo(&self, rhs: &Self::Rhs) -> Result<Value>;
+}
+
+trait BinaryOperator {
+    type Rhs;
+
+    fn eq(&self, rhs: &Self::Rhs) -> bool;
+    fn partial_cmp(&self, rhs: &Self::Rhs) -> Option<Ordering>;
+}
```

Comment:
1. This would be enough to be located in `value/binary_op.rs`
Then, in here we can use these from `binary_op` module.

2. `BinaryOperator` looks unnecessary
There already exists `PartialEq` and `PartialOrd` which are provided by Rust itself, we can simply use those rather than declaring custom `BinaryOperator` trait.

Reply:
2 will probably have a problem with the core implementation.
When implemented for i8 ...

![image](https://user-images.githubusercontent.com/27040963/144886878-b2fa4158-b957-418f-ba12-8a6a128ccbda.png)

Reply:
Those are supposed to work, hmm.. I don't get it why the conflict occurs but I've tested the codes below and there was no issue.

```rust
impl PartialEq<Value> for i8 {                                
    fn eq(&self, other: &Value) -> bool {                     
        match other {                                         
            I8(other) => self.eq(other),                      
            I64(other) => (*self as i64).eq(other),           
            F64(other) => (*self as f64).eq(other),           
            _ => false,                                       
        }                                                     
    }                                                         
}                                                             
                                                              
impl PartialOrd<Value> for i8 {                               
    fn partial_cmp(&self, other: &Value) -> Option<Ordering> {
        match other {                                         
            I8(other) => self.partial_cmp(other),             
            I64(other) => (*self as i64).partial_cmp(other),  
            F64(other) => (*self as f64).partial_cmp(other),  
            _ => None,                                        
        }                                                     
    }                                                         
}                                                             
```

Reply:
It seems to be because it is defined in Associate type. I edited